Начало поддержка выгрузки 8.3.7 в иерархическом формате.

pumbaEO opened the pull request:
На текущий момент сделал обратно совместимым, если необходима выгрузка стандартная в иерерхическом виде, то необходимо указать явно параметр -format . 
Дополнительно подправил выгрузку под linux, мелкая ошибка с регистром наименования базы данных.

Commit:
В linux создается автоматом в верхнем регистре и т.к. есть зависимость от регистра не проверялось на существование.

Commit:
Поддержка выгрузки 8.3.7 в иерархическом формате.

artbear commented on src/gitsync/src/unpack.os:
```diff
@@ -134,33 +134,54 @@
 		КонецЕсли;
 		
-		ДобавитьПереименование(Переименования,Файл.Имя,ИмяНовогоФайла);
+		ДобавитьПереименование(Переименования,ИмяФайлаДляПерименования,ИмяНовогоФайла);
```
Синтакс-ошибка 
вместо `ИмяФайлаДляПерименования` нужно `ИмяФайлаДляПереименования` и заменить по тексту процедуры

artbear commented on src/gitsync/src/unpack.os:
```diff
@@ -784,7 +805,7 @@
 // Основная функция синхронизации. Перемещает в git набор коммитов хранилища 1С
 //
-Процедура СинхронизироватьХранилищеКонфигурацийСГит(Знач КаталогРабочейКопии, Знач ФайлХранилища, Знач НачальнаяВерсия = 0, Знач КонечнаяВерсия = 0) Экспорт
+Процедура СинхронизироватьХранилищеКонфигурацийСГит(Знач КаталогРабочейКопии, Знач ФайлХранилища, Знач НачальнаяВерсия = 0, Знач КонечнаяВерсия = 0, Знач Формат = "hierarchical") Экспорт
```
Предлагаю не писать значение по умолчанию с магической строкой  `Знач Формат = "hierarchical"`
а сделать параметр Формат третьим параметром без значения по умолчанию и изменить код вызова этого метода

artbear commented on src/gitsync/src/unpack.os:
```diff
@@ -108,8 +108,8 @@
 //
-Процедура РазложитьМодули1СПоПапкамСогласноИерархииМетаданных(Знач КаталогПлоскойВыгрузки, Знач КаталогИерархическойВыгрузки) Экспорт
-	
+Процедура РазложитьМодули1СПоПапкамСогласноИерархииМетаданных(Знач КаталогПлоскойВыгрузки, Знач КаталогИерархическойВыгрузки, Знач Формат) Экспорт
```
Предлагаю вместо строки `Формат` использовать булевую переменную `ЭтоИерархическаяВыгрузка`
и преобразовывать из строки в булево в управляющем файле `gitsync.os'
код станет намного проще и понятнее.

artbear replied:
Тогда `магическое` знание `hierarchical|plain` останется только в `gitsync` и это будет хорошо :)

artbear replied:
@pumbaEO Что по моим комментариям скажешь?

Commit:
Исправление замечаний.

Убрал отладочные ненужные сообщения, перенес в переменную строку описания иерархии.

artbear commented:
@pumbaEO Пулл-реквест не работает с форматом plain :(
Нужно доработать:
1 Параметр формата не передается в функции `unpack.ВыгрузитьМодулиКонфигурации` и в вызов `/DumpConfigToFiles`
Нужно `"/DumpConfigToFiles """ + КаталогПлоскойВыгрузки + """ -format Plain"`
или `"/DumpConfigToFiles """ + КаталогПлоскойВыгрузки + """ -format "+Формат`
2 Для 1С важен регистр наименований `Hierarchical` или `Plain` (иначе будет ошибка выгрузки), а у тебя используется нижний регистр

pumbaEO commented:
Надо переделывать в принципе сам v8runner.os , добавлять туда поддержку
8.3.7 , мои правки - это мелкий фикс, который правильней было-бы в
отдельную ветку пока выделить, т.к. теряется полностью обратная
совместимость с 8.3.6... (хотя и виновата в данном случаи сама 1с,
сломавшая поведение по умолчанию).

9 декабря 2015 г., 12:55 пользователь Artur Ayukhanov <
notifications@github.com> написал:

> @pumbaEO https://github.com/pumbaEO Пулл-реквест не работает с форматом
> plain :(
> Нужно доработать:
> 1 Параметр формата не передается в функции
> unpack.ВыгрузитьМодулиКонфигурации и в вызов /DumpConfigToFiles
> Нужно "/DumpConfigToFiles """ + КаталогПлоскойВыгрузки + """ -format
> Plain"
> или "/DumpConfigToFiles """ + КаталогПлоскойВыгрузки + """ -format
> "+Формат
> 2 Для 1С важен регистр наименований Hierarchical или Plain (иначе будет
> ошибка выгрузки), а у тебя используется нижний регистр
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/EvilBeaver/oscript-library/pull/38#issuecomment-163166989
> .

## 

З повагою Сосна Євген,
mailto:shenja@sosna.zp.ua
skype:shenjasosna
+380933897103

artbear commented:
Я вернул обратную совместимость в https://github.com/EvilBeaver/oscript-library/commit/91ba744973a6834bba3c21a99fff01b2a8ba1966
До 8.3.7 параметр format 1С учитывать не будет, а для 8.3.7 уже будет.
у меня заработало.

Commit:
Merge branch 'develop' into linuxport

Commit:
Исправил проверку выгрузки на иерархическую.

EvilBeaver commented:
@pumbaEO @artbear парни, я уже немного запутался. Этот пулреквест можно сливать или он требует доработки?

pumbaEO commented:
Еще нет.
13 дек. 2015 г. 18:37 пользователь "Andrei Ovsiankin" <
notifications@github.com> написал:

> @pumbaEO https://github.com/pumbaEO @artbear
> https://github.com/artbear парни, я уже немного запутался. Этот
> пулреквест можно сливать или он требует доработки?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/EvilBeaver/oscript-library/pull/38#issuecomment-164268455
> .

EvilBeaver commented:
@artbear помоги прояснить: @pumbaEO сказал, что сливать пул-реквест пока рано, однако есть коммит 30c235f6622, в котором ты этот пулреквест влил. Но сам пулреквест все еще открыт. Я запутался.

artbear commented:
Я слил изменения, дополнил их и теперь новый код работает для старого умолчального (plain) формата.
Для нового формата я только ревьюил код, сам не тестил.

Но я не сливал последние правки из https://github.com/EvilBeaver/oscript-library/pull/38#commits-pushed-3c42619

И странно, что до сих пор числится много изменений, ведь я фактически все слил ранее :(
